internal/scorecard/alpha/scorecard.go

```diff
@@ -83,12 +99,15 @@ func selectTests(selector labels.Selector, tests []ScorecardTest) []ScorecardTes
 
 // runTest executes a single test
 // TODO once tests exists, handle the test output
-func runTest(test ScorecardTest) error {
+func runTest(o Options, test ScorecardTest) (result *v1.Pod, err error) {
```

Comment:
This implementation is going to make it difficult to write unit tests for the `RunTests` function. WDYT about making a `TestRunner` interface to use instead.

```go
type TestRunner
   func RunTest(context.Context, test Test, bundle *registry.Bundle) (*TestResult, error)
}
```

And then have the main struct (`Options` right now, maybe `Scorecard` based on another comment), have something like:
```go
type Scorecard struct {
	Config     Config
	Selector   labels.Selector		
	WaitTime   time.Duration
	Bundle     *registry.Bundle
	TestRunner TestRunner
}

var _ TestRunner = &PodTestBundleRunner{}

type PodTestRunner struct {
	Client         kubernetes.Interface
	Namespace      string
	ServiceAccount string
	SkipCleanup    bool
}

func (r *PodTestRunner) RunTest(ctx context.Context, test Test, bundle *registry.Bundle) (*TestResult, error) {
	return actuallyRunTheTest(ctx, test, bundle)
}

var _ TestRunner = &FakeTestRunner{}

type FakeTestRunner struct {
	TestResult *TestResult
	Error  error
}

func (r *FakeTestRunner) RunTest(ctx context.Context, test Test, bundle *registry.Bundle) (*TestResult, error) {
	return r.TestResult, r.Error
}
```

Then on the cmd CLI side, we can construct the TestRunner and set that field on the scorecard struct.

Reply:
interesting approach, but I'm wondering if we could address this in another PR since I don't think it impacts the test interface and I'd like to unblock the folks writing tests currently.  I think there is a story to rewrite the scorecard tests, maybe we could do this in that story?